src/npcmove.cpp

```diff
@@ -2198,7 +2205,17 @@ void npc::move_to( const tripoint &pt, bool no_bashing, std::set<tripoint> *nomo
         moved = true;
     } else if( g->m.passable( p ) ) {
         bool diag = trigdist && posx() != p.x && posy() != p.y;
-        moves -= run_cost( g->m.combined_movecost( pos(), p ), diag );
+        if( is_mounted() && mounted_creature ) {
```

Comment:
```suggestion
        if( is_mounted() ) {
```
Check for `mounted_creature` is already included.